builtin_string.go

```diff
@@ -371,6 +371,39 @@ func (r *Runtime) stringproto_match(call FunctionCall) Value {
 	panic(r.NewTypeError("RegExp matcher is not a function"))
 }
 
+func (r *Runtime) stringproto_matchAll(call FunctionCall) Value {
+	r.checkObjectCoercible(call.This)
+	regexp := call.Argument(0)
+	if regexp != _undefined && regexp != _null {
+		if isRegexp(regexp) {
+			if o, ok := regexp.(*Object); ok {
+				flags := o.Get("flags")
+				r.checkObjectCoercible(flags)
+				if !strings.Contains(flags.String(), "g") {
+					panic(r.NewTypeError("RegExp doesn't have global flag set"))
+				}
+			}
+		}
+		if matcher := toMethod(r.getV(regexp, SymMatchAll)); matcher != nil {
+			return matcher(FunctionCall{
+				This:      regexp,
+				Arguments: []Value{call.This},
+			})
+		}
+	}
+
+	rx := r.newRegExp(regexp, stringFromRune('g'), r.global.RegExpPrototype).self.(*regexpObject)
```

Comment:
```suggestion
	rx := r.newRegExp(regexp, asciiString("g"), r.global.RegExpPrototype).self.(*regexpObject)
```

Also, I think it's probably worth changing `newRegExp` (and its dependencies) to return `*regexpObject` instead of `*Object`. They cannot return anything else anyway (and if they do one day this line will panic) and it's easy enough to get the `*Object` from `*regexpObject`.